### ISSUE TITLE
[MEGA THREAD] Tracking issue for known kinds of corruptions

### ISSUE THREAD

**Issue by cbjeukendrup** [musescore/MuseScore#15739]
There are several reports of score corruptions, and I think it would be good to list them in one place and categorize them. So here we go:

- [x] https://github.com/musescore/MuseScore/issues/15185
    - https://github.com/musescore/MuseScore/issues/15360 `Crash Replication Test Site.mscz`
    - https://musescore.org/en/node/341414 [`Dum av dig.mscz`](https://musescore.org/sites/musescore.org/files/2023-01/Dum%20av%20dig_0.mscz)
    - https://musescore.org/en/node/341424 [`Gubben i lådan.mscz`](https://musescore.org/sites/musescore.org/files/2023-01/Gubben%20i%20l%C3%A5dan.mscz)
- [x] Corruption in main score; some notes in some measures too long; seems related to mysterious `<location>` tag??
    - https://github.com/musescore/MuseScore/issues/15519 `Clemens 4-handed.mscz`
    - https://musescore.org/en/node/341430 [`Crushdance.mscz`](https://musescore.org/sites/musescore.org/files/2023-01/Crushdance.mscz)
    - https://musescore.org/en/node/341127#comment-1163573 `The_Physical_Impossibility_bug_report_samples.mscz`
    - https://musescore.org/en/node/340301 https://musescore.org/en/node/340353 [`Mechanically Separated Chicken.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/Mechanically%20Separated%20Chicken_1.mscz)
    - https://musescore.org/en/node/340301#comment-1159814 [`Piece for Cello 1.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/Piece%20for%20Cello%201_0.mscz)
    - https://musescore.org/en/node/340552 [`StringQuartet1.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/StringQuartet1_0.mscz)
    - https://musescore.org/en/node/340126 [`866.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/866.mscz)
    - https://musescore.org/en/node/339824 [`A Nightingale Sang in Berkeley Square_2.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/A%20Nightingale%20Sang%20in%20Berkeley%20Square_2.mscz)
    - https://musescore.org/en/node/339686 [`Piel Canela_Bbmaj__ls04.MS4_.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/Piel%20Canela_Bbmaj__ls04.MS4_.mscz)
    - Seemingly https://musescore.org/en/node/339315 too, but that one should be taken with a grain of salt because it's from 1.x
    - https://musescore.org/en/node/338567 [`Momentum_-_Cosmos_G..mscz`](https://musescore.org/sites/musescore.org/files/2022-12/Momentum_-_Cosmos_G..mscz)
    - maybe this one too: https://musescore.org/en/node/339165 [`Missa_Rearranged V4.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/Missa_Rearranged%20V4.mscz) (it looks like the corruption is in the parts, but then breaks the main score on read; the parts corruption looks like it's caused by the first corruption mentioned)
    - https://github.com/musescore/MuseScore/issues/15689 [`Polyamateur 4 MS4.mscz.zip`](https://github.com/musescore/MuseScore/files/10333069/Polyamateur.4.MS4.mscz.zip)
- [x] ~Corruption in main score; tuplets seem somehow lost~
    - https://github.com/musescore/MuseScore/issues/15723 `The Merry-Go-Round of Life – Joe Hisaishi (FULL_ Orchestral Transcription).mscz`
- [x] Sort of corruption in main score due to actions with measure repeat: https://github.com/musescore/MuseScore/issues/15844#issuecomment-1382866721
    - steps:
        1. Create score with measure containing notes, another measure containing measure repeat
        2. Copy contents from measure containing notes to measure repeat measure
        3. To see the corruption, press Delete on the measure that previously had the measure repeat: you see a strange empty measure now
- [x] ~#15490 which is currently in need of info~
- [x] https://github.com/musescore/MuseScore/issues/15679
- [ ] #16096 which is currently in need of info
- [ ] #16337
    An unclear one: https://musescore.org/en/node/343454 (the score is not detected/reported as corrupted, but the log output clearly shows that it is)
    (it's a single one so far, but quite a strange one, since it looks like staff indices somehow got out of sync between the master score and the part score)
- [ ] Zip file corrupted -> no recovery possible
    - https://github.com/musescore/MuseScore/issues/15113 `Bossa_Lemanica.mscz`
    - https://discord.com/channels/818804595450445834/818804809784229898/1057744302765256874 (I know discord links are useless but it would be useless anyway, since a file full of nulls is not very interesting)
    - https://musescore.org/en/node/342465 file became 0KB
- [ ] A very peculiar case where the main mscx file is missing from the mscz file but the rest is there
    https://musescore.org/en/node/340829#comment-1163670 [`.Orquestrado.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/.Orquestrado.mscz)
    (Oh wait, maybe this is because this person had the great idea of starting the filename with a ".")
- [ ] https://github.com/musescore/MuseScore/issues/16338
    Another very peculiar case where the main mscx file is not at root level in the mscz file, but in "Users/perfc/Downloads" inside the mscz file
    https://musescore.org/en/node/340775 [`Carol of the Bells.mscz`](https://musescore.org/sites/musescore.org/files/2022-12/Carol%20of%20the%20Bells.mscz)
   This has come up another time, namely someone reported it to Martin directly

**Comment by cbjeukendrup** [comment]
If you find any corrupt score in MS4 or a new report thereof, please comment here, and I'll add it to the list!

**Comment by cbjeukendrup** [comment]
After one evening of trawling through the musescore.org issue tracker, it has become obvious that the `<location>`-related (?) one is by far the biggest. From looking at the scores, nothing becomes immediately clear, except that in a lot of cases a `<location>` tag is present around the location of the corruption. 
However, I'm starting to suspect that the cause of these corruptions is in fact the same as the first corruption in the list. Initially, the main score would still be intact, and only the parts would be broken; but after some saves and especially re-opens, the corruptions propagate to the main score. So far, this is supported by the observation that in one case, the score opened without any corruptions after I had extracted the main mscx file outside of the mscz, effectively deleting the part scores. I'll try this with more scores.

**Comment by Tantacrul** [comment]
Excellent. Thanks @cbjeukendrup

**Comment by Tantacrul** [comment]
I know @vpereverzev thinks that the majority, if not all these issues are fixed. Can you confirm Casper? Might make sense to actually just assign incomplete issues here individually and remove the megathread.

**Comment by cbjeukendrup** [comment]
I also suspect that the cause of the first two points is fixed, but it is difficult to confirm that this was indeed the cause in all those cases. Maybe the best way is to wait a while and watch the forums to see if indeed no new corruptions are reported.

But yeah, for the other cases, there doesn't really seem to be a pattern, and they have been reported only once, so it makes sense to handle them individually. I don't think they are new in MS4 either, actually.

(Except the last two; those are caused by the new file format, and may be related in a way.)

**Comment by cbjeukendrup** [comment]
By the way, here is a mega mega thread about those mysterious ZIP file corruptions (where the file either becomes 0 bytes, or becomes filled with only zeros and is therefore useless): https://musescore.org/en/node/271185

So that's really not new... but also not fixed by the changes to the ZIP writer in MS4.

**Comment by cbjeukendrup** [comment]
Good news: since 4.0.1 I have seen no reports anymore of new corrupted files. I think this is statistically significant, so that we can conclude that the first two big ones can be marked as fixed.

Bad news: there are two new ones (I'll add them to the issue description)
- for some people, spanners (hairpins, voltas, lines.....) are missing in part scores
    - https://musescore.org/en/node/341765
    - https://musescore.org/en/node/343502
- there's this corrupted case: https://musescore.org/en/node/343454 (the score is not detected/reported as corrupted, but the log output clearly shows that it is)
(it's a single one so far, but quite a strange one, since it looks like staff indices somehow got out of sync between the master score and the part score)

**Comment by cbjeukendrup** [comment]
I've now created distinct issues for the remaining ones, and labeled them with a new `corruption` label to keep track of them. So I think we're ready to close this thread. When a new corruption is found, users can create a separate issue for it, and if it turns out to be caused by the same thing as an existing report, we can just link them together and close one of them, like we always do.

**Comment by DmitryArefiev** [comment]
Hard to test all of those issue. Moving to **Done - Tested Again**
